pkg/porter/generateManifest.go

```diff
@@ -87,7 +88,23 @@ func (p *Porter) generateInternalManifest(ctx context.Context, opts BuildOptions
 		}
 		ref := fmt.Sprintf("%s:%s", img.Repository, imgTag)
 
-		digest, err := p.Registry.PullImage(ctx, ref)
+		imgSummary, err := p.Registry.GetCachedImage(ctx, ref)
```

Comment:
This alters the behavior of this function, because it switches from always pulling (which will be speedy regardless if its cached) to possibly short circuiting and returning an older cached image.

Let's go back to what it did before. So it should call PullImage, and then follow up with GetCachedImage.